consensus/propagation/have_wants.go

```diff
@@ -106,7 +104,7 @@ func (blockProp *Reactor) handleHaves(peer p2p.ID, haves *proptypes.HaveParts, b
 		},
 	}
 
-	if !p2p.SendEnvelopeShim(p.peer, e, blockProp.Logger) { //nolint:staticcheck
+	if !p2p.TrySendEnvelopeShim(p.peer, e, blockProp.Logger) { //nolint:staticcheck
```

Comment:
since we're not making sure the message is sent, does it make sense to increase the request limit to 2? for additional redundancy

Reply:
imo I don't think it makes sense. increasing redundancy to 2 still doesn't solve the problem, as nodes with full queues will still miss the msg. it will also increase bandwidth usage by 2x

this should be handled properly when we ensure that there are no cases where the compact block is sent before a have